aya/src/bpf.rs

```diff
@@ -345,8 +345,12 @@ impl<'a> BpfLoader<'a> {
 
         let btf_fd = if let Some(features) = &FEATURES.btf {
             if let Some(btf) = obj.fixup_and_sanitize_btf(features)? {
-                // load btf to the kernel
-                Some(load_btf(btf.to_bytes())?)
+                match load_btf(btf.to_bytes()) {
+                    Ok(btf_fd) => Some(btf_fd),
+                    // Only report an error here if the BTF is truely needed, otherwise proceed without.
+                    Err(err) if obj.kernel_needs_btf() => return Err(BpfError::BtfError(err)),
```

Comment:
I think it would be better if we iterated obj.program here, looking at
program.section and deciding based on that whether we actually need BTF or not.
That way we don't need to set new state (`has_struct_ops`) and we don't need
`kernel_needs_btf` either, which doesn't seem like something that belongs to
`aya-obj`. And we need to check for all programs that require BTF, not only
struct ops. 

Also, I think we should only ignore the error if the user hasn't explicitly
called `BpfLoader::btf()` - if they explicitly asked for BTF to be loaded, they
might want BTF for other reasons (eg line info), and we shouldn't silently fail
to load it. In that case we should return the error and let callers decide
whether they want to try loading again with `btf(None)`.